src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -2,21 +2,37 @@
 
 import java.util.Optional;
 import mate.academy.hibernate.relations.dao.ActorDao;
+import mate.academy.hibernate.relations.exception.DataProcessingException;
 import mate.academy.hibernate.relations.model.Actor;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
 
 public class ActorDaoImpl extends AbstractDao implements ActorDao {
+
     public ActorDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        try (Session session = factory.openSession()) {
```

Comment:
Transaction implemented in wrong way. 
It will not work. 

P.S. Where is your rollback?

Reply:
Transaction still does not work 

when you call `session.getTransaction` 
session could return diff transactions, so it wont work as you expect.